Libraries/libbom/Sources/bom.c

```diff
@@ -33,21 +34,30 @@ _bom_alloc(struct bom_context_memory memory)
 
     context->memory = memory;
     context->iteration_count = 0;
+    context->preallocated_index_count = 0;
 
     return context;
 }
 
 struct bom_context *
 bom_alloc_empty(struct bom_context_memory memory)
 {
+  return bom_alloc_size(memory, 2);
+}
+
+struct bom_context *
+bom_alloc_size(struct bom_context_memory memory, uint32_t index_count)
+{
     struct bom_context *context = _bom_alloc(memory);
     if (context == NULL) {
         return NULL;
     }
 
+    context->preallocated_index_count = index_count;
+
     size_t header_size = sizeof(struct bom_header);
     size_t index_size = sizeof(struct bom_index_header);
-    size_t freelist_size = sizeof(struct bom_index_header) + sizeof(struct bom_index) * 2;
```

Comment:
IIRC, the value here wasn't arbitrary — I think I saw that some tools didn't handle files with a free list of size less than `2`. So it might be worth using `(2 + index_count)` here and pass `0` as the default, so you can't accidentally make an invalid file.